### PR TITLE
TINY-9839: Adding editable root API

### PR DESCRIPTION
**Pull request by spocke**
Related Ticket: TINY-9839

Description of Changes:
* Adds the new editable root API according to spec
* Adds the new newdocument_content so that you can set default content when the root is non-editable

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] Docs ticket created (if applicable)

GitHub issues (if applicable):

### CODE REVIEW

File: modules/tinymce/CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - New `accordion` and `accordion-toggle` icons. #TINY-9789
 - New `details_initial_state` and `details_serialized_state` options. #TINY-9732
 - New `init_content_sync` option that initializes the editor iframe using `document.write` instead of `srcdoc`. #TINY-9818
+- New `newdocument_content` option that changes the content that will be set when using the `New document` button. #TINY-9839
+- New `editable_root` option that can be set to `false` to prevent editing of the editors root element. #TINY-9839
+- New `editor.setEditableRoot` API that sets the editable state of the editor root element. #TINY-9838
+- New `editor.hasEditableRoot` API that returns `true` or `false` depending on the editable state of the editor root element. #TINY-9838
+- New `EditableRootStateChange` event that gets dispatched when the state of the editable root is changed. #TINY-9838
```
Comment (vpyshnenko) [inline modules/tinymce/CHANGELOG.md:24]: Wrong ticket number at the end of change log entry

Comment (spocke) [reply modules/tinymce/CHANGELOG.md:24]: Good catch. :+1:

File: modules/tinymce/src/core/test/ts/browser/EditableRootTest.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
+import { context, describe, it } from '@ephox/bedrock-client';
+import { TinyHooks } from '@ephox/wrap-mcagar';
+import { assert } from 'chai';
+
+import Editor from 'tinymce/core/api/Editor';
+
+describe('browser.tinymce.core.EditableRootTest', () => {
+  const assertContentEditableState = (editor: Editor, expectedState: boolean) => {
+    assert.equal(editor.getBody().contentEditable === 'true', expectedState);
```
Comment (vpyshnenko) [inline modules/tinymce/src/core/test/ts/browser/EditableRootTest.ts:9]:
```suggestion
    assert.equal(editor.getBody().isContentEditable, expectedState);
```

File: modules/tinymce/src/core/test/ts/browser/EditableRootTest.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  };
+
+  const assertEditableState = (editor: Editor, expectedState: boolean) => {
```
Comment (vpyshnenko) [inline modules/tinymce/src/core/test/ts/browser/EditableRootTest.ts:12]:
```suggestion
  const assertRootEditableState = (editor: Editor, expectedState: boolean) => {
```

File: modules/tinymce/src/core/test/ts/browser/EditableRootTest.ts [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+      assertEditableState(hook.editor(), false);
+      assert.equal(initialContentEditableState, 'inherit', 'Should not be editable early in the init process');
+    });
+  });
+
+  context('editable_root: default', () => {
+    const hook = TinyHooks.bddSetupLight<Editor>({
+      base_url: '/project/tinymce/js/tinymce'
+    }, [], true);
+
```
Comment (vpyshnenko) [inline modules/tinymce/src/core/test/ts/browser/EditableRootTest.ts:40]: maybe we need to reset the state for the tests do not affect each other
`afterEach(() => {
    editor.setEditableRoot(true);
})`

**Review by vpyshnenko — APPROVED** [review]

File: modules/tinymce/src/core/test/ts/browser/EditableRootTest.ts (hunk 4 above)
Comment (spocke) [reply modules/tinymce/src/core/test/ts/browser/EditableRootTest.ts:40]: I think I reset the current test by setting back the state I guess we could use a after if resetting the state becomes a problem and if there is more than just this simple test.

**Review by vpyshnenko — APPROVED** [review]

**Review by lorenzo-pomili — APPROVED** [review]

**Pull request merged**